src/github.com/couchbase/sync_gateway/base/bucket_gocb.go

```diff
@@ -476,13 +496,45 @@ func isRecoverableGoCBError(err error) bool {
 	if err == nil {
 		return false
 	}
+
+	goCBErr := GoCBErrorType(err)
+
+	_, ok := recoverableGoCBErrors[goCBErr]
+
+	return ok
+}
+
+// GoCB error types - workaround until gocb has public error type lookup support
```

Comment:
Somewhat tangential to this PR, but did you file a ticket for the GOCB team to make their error types public?